pandas/core/internals/construction.py

```diff
@@ -381,6 +399,7 @@ def extract_index(data):
                 )
 
             if have_series:
+                assert index is not None  # for mypy
```

Comment:
is this code (and above) reachable? if have_raw_arrays is True, have_series and have_dicts is False?